addons/mail/models/res_partner.py

```diff
@@ -248,6 +248,7 @@ def _to_store(self, store: Store, /, *, fields=None, main_user_by_partner=None):
                     data["isAdmin"] = main_user._is_admin()
                 if "notification_type" in fields:
                     data["notification_preference"] = main_user.notification_type
+                data["signature"] = main_user.signature
```

Comment:
Don't add this for all partners all the time. This needs to be conditioned to receiving the field as param.
And the field should only be set for the current user in `_init_store_data`.

Reply:
Hello @seb-odoo Sorry to bother you. I'm not sure to understand what you want me to do. Thinking about it, I assume it should be something like this:

In the `_init_store_data` function of `res_users.py`, I have to add a new entry `signature` to the requested field list.

```py
"self": Store.one(
    self.env.user.partner_id,
    fields=[
        "active",
        "isAdmin",
        "name",
        "notification_type",
        "signature",
        "user",
        "write_date",
    ],
    main_user_by_partner={self.env.user.partner_id: self.env.user},
),
```

Then, in the `_to_store` function of `res_partner.py`, when iterating over the partners, I have to exclude `signature` from the `_read_format` as that field does not exist on the `res.partner` model.

```py
for partner in self:
    data = partner._read_format(
        [
            field
            for field in fields
            if field
            not in ["country", "display_name", "isAdmin", "notification_type", "signature", "user"]
        ],
        load=False,
    )[0]
```

Then, I have to add a new condition to add a `signature` entry when the `signature` field is requested (i.e: when there is a `signature` key in the `fields` variable).

```py
if "signature" in fields:
    data["signature"] = main_user.signature
```

Is it what you mean ?

Reply:
Yes exactly, the result of this change should be visible in tests: most of them should stay unchanged in the PR.
The only test to change should be the one with the init data.

Note: it's unfortunate to have to exclude the field from partner and then add it manually, I plan to properly split between user and partner data at some point but the current solution should work just fine.